src/main/java/org/broadinstitute/hellbender/tools/spark/sv/FindBadGenomicKmersSpark.java

```diff
@@ -86,54 +86,51 @@ public static void writeKmersToFile( final File kmersFile, final List<SVKmer> km
+     * K-1 base overlap between successive chunks. (I.e., for K=63, the last 62 bases in chunk n match the first 62
+     * bases in chunk n+1) so that we don't miss any kmers due to the chunking -- we can just kmerize each record
+     * independently.
      */
-    private static File writeRefAsOverlappingRecords( final ReferenceMultiSource ref,
-                                                      final GCSOptions gcsOptions,
-                                                      final SAMSequenceDictionary readsDict ) {
-        final File refSequenceFile;
-        try {
-            refSequenceFile = File.createTempFile("genomic_sequence", "txt");
-            refSequenceFile.deleteOnExit();
-            try ( final OutputStream os = new BufferedOutputStream(new FileOutputStream(refSequenceFile)) ) {
-                final SAMSequenceDictionary dict = ref.getReferenceSequenceDictionary(readsDict);
-                if ( dict == null ) throw new GATKException("No reference dictionary available.");
+    private static JavaRDD<byte[]> getRefRDD( final JavaSparkContext ctx,
+                                              final ReferenceMultiSource ref,
+                                              final PipelineOptions options,
+                                              final SAMSequenceDictionary readsDict ) {
+        final SAMSequenceDictionary dict = ref.getReferenceSequenceDictionary(readsDict);
+        if ( dict == null ) throw new GATKException("No reference dictionary available.");
 
-                final int effectiveRecLen = REF_RECORD_LEN - SVConstants.KMER_SIZE + 1;
-                for ( final SAMSequenceRecord rec : dict.getSequences() ) {
-                    final String seqName = rec.getSequenceName();
-                    final int seqLen = rec.getSequenceLength();
-                    final SimpleInterval interval = new SimpleInterval(seqName, 1, seqLen);
-                    final byte[] bases = ref.getReferenceBases(gcsOptions, interval).getBases();
-                    for ( int start = 0; start < seqLen; start += effectiveRecLen ) {
-                        os.write(bases, start, Math.min(REF_RECORD_LEN, seqLen - start));
-                        os.write('\n');
-                    }
+        final int effectiveRecLen = REF_RECORD_LEN - SVConstants.KMER_SIZE + 1;
+        final List<byte[]> sequenceChunks = new ArrayList<>();
+        for ( final SAMSequenceRecord rec : dict.getSequences() ) {
+            final String seqName = rec.getSequenceName();
+            final int seqLen = rec.getSequenceLength();
+            final SimpleInterval interval = new SimpleInterval(seqName, 1, seqLen);
+            try {
+                final byte[] bases = ref.getReferenceBases(options, interval).getBases();
+                for ( int start = 0; start < seqLen; start += effectiveRecLen ) {
+                    sequenceChunks.add(Arrays.copyOfRange(bases, start, Math.min(start+REF_RECORD_LEN, seqLen)));
```

Comment:
Ah.  I should've been able to figure that out.
I guessed that that might have some efficiency problems, especially since I'm accessing in an unusual pattern (the K-1 base overlap).  This way all the index manipulation, caching, and whatever else is going on in that method happens just once per contig.  If you're pretty sure that querying 10K chunks with a K-1 base chunk overlap will be reasonably efficient on all the implementations of this interface I'll just change it over and won't look back or second-guess you.